### PR TITLE
Changes to the clip PDF flow and notifications

### PR DESCRIPTION
**Pull request by ipdutta**
Changes included -
1) PDF XMLHttpRequest to fetch
2) window.location.href to self.location.href
3) alert to notifications in background script

### CODE REVIEW

File: src/scripts/extensions/injectHelper.ts [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 	private static uninjectableUrlRegexes: RegExp[] = [
 		/^about:/
 	];
 
 	public static alertUserOfUnclippablePage() {
-		alert(Localization.getLocalizedString("WebClipper.Error.CannotClipPage"));
+		WebExtension.browser.notifications.create({
+			type: 'basic',
+			iconUrl: '/icons/icon-48.png',
+			title: 'Cannot Clip Page',
```
Comment (ArpanKundu97) [inline src/scripts/extensions/injectHelper.ts]: Shall we localize the string for title as well?

File: src/scripts/contentCapture/pdfScreenshotHelper.ts [hunk 2]
```diff
@@ -29,42 +30,40 @@ export class PdfScreenshotHelper {
...
 			};
 
-			request.onload = () => {
-				if (request.status === 200 && request.response) {
-					let arrayBuffer = request.response;
 
+			fetch(url)
+			.then(response => {
+				if (!response.ok) {
+					errorCallback(ErrorUtils.createRequestErrorObject(response, OneNoteApi.RequestErrorType.UNEXPECTED_RESPONSE_STATUS));
```
Comment (ArpanKundu97) [inline src/scripts/contentCapture/pdfScreenshotHelper.ts:47]: `createRequestErrorObject` now returns a Promise. Should we add a `then` block here?

File: src/scripts/contentCapture/pdfScreenshotHelper.ts [hunk 3]
```diff
@@ -29,42 +30,40 @@ export class PdfScreenshotHelper {
...
-			};
-
-			request.send();
+					PdfScreenshotHelper.getPdfScreenshotResult(new Uint8Array(arrayBuffer))
+						.then(pdfScreenshotResult => {
+							pdfScreenshotResult.byteLength = arrayBuffer.byteLength;
+							resolve(pdfScreenshotResult);
+						})
+				})
+			.catch(() => {
```
Comment (ArpanKundu97) [inline src/scripts/contentCapture/pdfScreenshotHelper.ts:61]: Should we create a request error object here as well?

File: src/scripts/extensions/injectHelper.ts (hunk 1 above)
Comment (ipdutta) [reply src/scripts/extensions/injectHelper.ts]: Done

File: src/scripts/contentCapture/pdfScreenshotHelper.ts (hunk 2 above)
Comment (ipdutta) [reply src/scripts/contentCapture/pdfScreenshotHelper.ts:47]: My rationale behind this is that we pass the Promise object directly into the errorCallback function and have a then block inside it to resolve the promise. Let me know if you see a problem with this approach.

File: src/scripts/contentCapture/pdfScreenshotHelper.ts (hunk 3 above)
Comment (ipdutta) [reply src/scripts/contentCapture/pdfScreenshotHelper.ts:61]: The response object will not be available in the catch block hence not able to create a request error object here

File: src/scripts/contentCapture/pdfScreenshotHelper.ts (hunk 2 above)
Comment (ArpanKundu97) [reply src/scripts/contentCapture/pdfScreenshotHelper.ts:47]: Got it, thanks!

**Review by ArpanKundu97 — APPROVED** [review]

**Pull request merged**